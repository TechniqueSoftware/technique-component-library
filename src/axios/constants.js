import { IS_PROD } from '@global/env'

const API_V3_DOMAIN = window.IS_PROD ? 'https://api.club-os.io' : 'https://test-api.club-os.io'
const API_V3_ACCESS_TOKEN_NAME = 'apiV3AccessToken'
const API_V3_ID_TOKEN_NAME = 'apiV3IdToken'

// const CONTEXT_PATH = IS_PROD ? window.CONTEXT_PATH : 'https://test-api.club-os.io'
const CONTEXT_PATH = IS_PROD ? window.CONTEXT_PATH : window.location.origin

export {
  API_V3_DOMAIN,
  API_V3_ACCESS_TOKEN_NAME,
  API_V3_ID_TOKEN_NAME,
  CONTEXT_PATH
}
