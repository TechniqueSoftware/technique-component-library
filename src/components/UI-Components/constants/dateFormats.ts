// Learn about endians: https://en.wikipedia.org/wiki/Date_format_by_country

export const DATE_FORMAT_ISO = 'YYYY-MM-DD';

export const DATE_FORMAT_MIDDLE_ENDIAN = 'MM/DD/YYYY';

export const DATE_FORMAT_LOCAL_DATETIME = 'YYYY-MM-DDTHH:mm:ss.sss';

export const DATE_FORMAT_FULL = 'dddd, MMMM Do, YYYY';
