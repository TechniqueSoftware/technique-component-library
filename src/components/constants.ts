export const MODULE_NAME = {
  HEADERS: 'Headers',
  DIALOGS: 'Dialogs',
  TABS: 'Tabs',
  BUTTONS: 'Buttons',
  PROGRESS: 'Progress',
  CARDS: 'Cards',
  MULTISELECT: 'MultiSelect',
  INPUTS: 'Inputs',
  ICON: 'Icon',
  TABLES: 'Tables',
  SNACKBARS: 'Snackbars',
  NOTIFICATIONS: 'Notifications',
  SURFACES: 'Surfaces',
  BREADCRUMBS: 'Breadcrumbs',
  ERRORS:'Errors',
};
