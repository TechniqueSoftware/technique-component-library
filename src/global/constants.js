export const COLORS = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  SUCCESS: 'success',
  DANGER: 'danger',
  WARNING: 'warning',
  INFO: 'info',
  LIGHT: 'light',
  DARK: 'dark',
  GREY: 'grey'
}

export const SIZES = {
  // XS: 'xs',
  SM: 'sm',
  MD: 'md',
  LG: 'lg'
  // XL: 'xl'
}

export const BUTTON_VARIANT = {
  SOLID: 'solid',
  OUTLINE: 'outline',
  LINK: 'link',
  GHOST: 'ghost'
}

export const INPUT_TYPE = {
  EMAIL: 'email',
  NUMBER: 'number',
  DATE: 'date',
  PASSWORD: 'password'
}

export const DEVICES = {
  DESKTOP: 1024,
  TABLET: 768,
  MOBILE: 0
}
