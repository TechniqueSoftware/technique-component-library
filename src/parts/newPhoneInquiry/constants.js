export const EVENT_ICONS = {
  1: 'MEETING',
  2: 'TRAINER',
  3: 'GROUP_CLASS',
  4: 'SALES_LEAD',
  5: 'ORIENTATION',
  6: 'ASSESSMENT',
  7: 'GROUP_TRAINING',
  8: 'GROUP_CLASS'
}
