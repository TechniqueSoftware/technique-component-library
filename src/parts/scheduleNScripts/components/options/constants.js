export const SCRIPT_CONTACT_TYPES = {
  EMAIL: 2,
  SMS: 3,
  NONE: 4
}
