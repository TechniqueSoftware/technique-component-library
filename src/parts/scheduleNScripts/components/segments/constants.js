export const FOLLOWUP_STATUS = {
  APPT_BOOKED: 2,
  MEMBER: 6,
  WEB_LEAD: 11
}
