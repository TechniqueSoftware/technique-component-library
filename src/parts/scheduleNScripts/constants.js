export const TP_ACTION_TYPES = {
  CALL: 1,
  EMAIL: 2,
  TEXT: 3
}

export const SCRIPT_TABNAME = [{
  id: '1',
  name: 'Call',
  className: 'BsTelephone'
}, {
  id: '2',
  name: 'Email',
  className: 'BsEnvelope'
}, {
  id: '3',
  name: 'Text',
  className: 'BsChatDots'
}]

export const SEGMENT_TABNAME = [
  {
    id: '0',
    name: 'Manage Segments'
  },
  {
    id: '1',
    name: 'Copy Segments'
  }
]

export const SNACKBAR_MESSAGES = {
  NEW_SUCCESS: 'Touchpoint has been created successfully!',
  EDIT_SUCCESS: 'Touchpoint has been updated!',
  DELETE_SUCCESS: 'Touchpoint has been deleted successfully!',
  SCRIPT_SUCCESS: 'Script has been saved successfully!',
  SEGMENT_NEW: 'Segment has been created successfully!',
  SEGMENT_EDIT: 'Changes for this segment have been saved successfully!',
  SEGMENT_DELETE: 'Segment has been deleted successfully!'
}

export const STATUSES_WITH_SEGMENTS = [
  { type: 1, id: 11, attributeType: 'Marketing Source', attributeId: 'marketing' }, // Prospecting -> Web Lead
  { type: 1, id: 2, attributeType: 'Marketing Source', attributeId: 'marketing' }, // Prospecting -> Appt Booked
  { type: 1, id: 6, attributeType: 'Membership Type', attributeId: 'membership' } // Member -> Member
]

export const PERMISSION_KEYS = {
  SCHEDULES: {
    1: 'canEditMemberSalesSchedules',
    2: 'canEditPtSchedules',
    3: 'canEditStudioSchedules'
  },
  SCRIPTS: {
    1: 'canEditMemberSalesScripts',
    2: 'canEditPtScripts',
    3: 'canEditStudioScripts'
  }
}
