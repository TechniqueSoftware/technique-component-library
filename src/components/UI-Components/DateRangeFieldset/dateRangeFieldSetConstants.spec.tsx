import {
  textCampaignDateRangeOptions,
  analyticsDateRangeOptions,
  DateRangeOption,
  emailCampaignDateRangeOptions
} from './dateRangeFieldSetConstants';
import { DATE_FORMAT_ISO, DATE_FORMAT_MIDDLE_ENDIAN } from '../constants/dateFormats';
import moment from 'moment';

describe('dateRangeFieldSetConstants', () => {

  function testDateRangeOptions(dateRangeOptions: DateRangeOption[]) {
    const results = dateRangeOptions.map((option) => {
      let calculateDateRange: { start: moment.Moment; end: moment.Moment };
      if (option.calculateDateRange) {
        calculateDateRange = option.calculateDateRange(moment('2019-01-20 06:66:66Z', DATE_FORMAT_ISO));
      }

      let formattedRange;
      if (calculateDateRange) {
        const { start, end } = calculateDateRange;
        formattedRange = {
          start: start && start.format(DATE_FORMAT_MIDDLE_ENDIAN),
          end: end && end.format(DATE_FORMAT_MIDDLE_ENDIAN)
        };
      }
      return [option.label, formattedRange];
    });
    expect(results).toMatchSnapshot();
  }

  test('textCampaignDateRangeOptions calculated date range should match snapshot', () => {
    testDateRangeOptions(textCampaignDateRangeOptions);
  });

  test('emailCampaignDateRangeOptions calculated date range should match snapshot', () => {
    testDateRangeOptions(emailCampaignDateRangeOptions);
  });

  test('analyticsDateRangeOptions calculated date range should match snapshot', () => {
    testDateRangeOptions(analyticsDateRangeOptions);
  });
});
