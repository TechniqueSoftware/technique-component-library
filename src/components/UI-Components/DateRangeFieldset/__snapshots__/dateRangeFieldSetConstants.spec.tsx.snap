// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dateRangeFieldSetConstants analyticsDateRangeOptions calculated date range should match snapshot 1`] = `
Array [
  Array [
    "Yesterday",
    Object {
      "end": "01/19/2019",
      "start": "01/19/2019",
    },
  ],
  Array [
    "This Week",
    Object {
      "end": "01/26/2019",
      "start": "01/20/2019",
    },
  ],
  Array [
    "Last Week",
    Object {
      "end": "01/19/2019",
      "start": "01/13/2019",
    },
  ],
  Array [
    "This Month",
    Object {
      "end": "01/31/2019",
      "start": "01/01/2019",
    },
  ],
  Array [
    "Last Month",
    Object {
      "end": "12/31/2018",
      "start": "12/01/2018",
    },
  ],
  Array [
    "Custom",
    undefined,
  ],
  Array [
    "This Year",
    Object {
      "end": "12/31/2019",
      "start": "01/01/2019",
    },
  ],
  Array [
    "Last Year",
    Object {
      "end": "12/31/2018",
      "start": "01/01/2018",
    },
  ],
]
`;

exports[`dateRangeFieldSetConstants emailCampaignDateRangeOptions calculated date range should match snapshot 1`] = `
Array [
  Array [
    "Yesterday",
    Object {
      "end": "01/19/2019",
      "start": "01/19/2019",
    },
  ],
  Array [
    "This Week",
    Object {
      "end": "01/26/2019",
      "start": "01/20/2019",
    },
  ],
  Array [
    "Last Week",
    Object {
      "end": "01/19/2019",
      "start": "01/13/2019",
    },
  ],
  Array [
    "This Month",
    Object {
      "end": "01/31/2019",
      "start": "01/01/2019",
    },
  ],
  Array [
    "Last Month",
    Object {
      "end": "12/31/2018",
      "start": "12/01/2018",
    },
  ],
  Array [
    "Custom",
    undefined,
  ],
  Array [
    "All Time",
    Object {
      "end": "01/20/2019",
      "start": "04/01/2018",
    },
  ],
]
`;

exports[`dateRangeFieldSetConstants textCampaignDateRangeOptions calculated date range should match snapshot 1`] = `
Array [
  Array [
    "Yesterday",
    Object {
      "end": "01/19/2019",
      "start": "01/19/2019",
    },
  ],
  Array [
    "This Week",
    Object {
      "end": "01/26/2019",
      "start": "01/20/2019",
    },
  ],
  Array [
    "Last Week",
    Object {
      "end": "01/19/2019",
      "start": "01/13/2019",
    },
  ],
  Array [
    "This Month",
    Object {
      "end": "01/31/2019",
      "start": "01/01/2019",
    },
  ],
  Array [
    "Last Month",
    Object {
      "end": "12/31/2018",
      "start": "12/01/2018",
    },
  ],
  Array [
    "Custom",
    undefined,
  ],
  Array [
    "All Time",
    Object {
      "end": "01/20/2019",
      "start": "01/01/2018",
    },
  ],
  Array [
    "This Year",
    Object {
      "end": "12/31/2019",
      "start": "01/01/2019",
    },
  ],
  Array [
    "Last Year",
    Object {
      "end": "12/31/2018",
      "start": "01/01/2018",
    },
  ],
]
`;
